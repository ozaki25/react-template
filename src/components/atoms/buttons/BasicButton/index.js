import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Color from 'src/constants/Color';

const BasicButton = ({
  children, onClick, disabled, small, xsmall, red,
}) => (
  <Button onClick={onClick} disabled={disabled} small={small} xsmall={xsmall} red={red}>
    {children}
  </Button>
);

const Button = styled.button`
  background-color: ${({ red }) => (red ? Color.brand : Color.darkButton)};
  border-radius: 4px;
  color: ${Color.white};
  font-size: 16px;
  font-weight: bold;
  height: 44px;
  width: ${({ small, xsmall }) => (xsmall ? '150px' : small ? '250px' : '300px')};
`;

BasicButton.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  small: PropTypes.bool,
  xsmall: PropTypes.bool,
  red: PropTypes.bool,
};

BasicButton.defaultProps = {
  onClick: null,
  disabled: false,
  small: false,
  xsmall: false,
  red: false,
};

export default BasicButton;
