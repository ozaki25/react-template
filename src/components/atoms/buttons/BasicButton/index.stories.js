import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { boolean, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import BasicButton from '.';

const stories = storiesOf('atoms/buttons/BasicButton', module);

stories.add(
  'nomal',
  withInfo('BasicButton/nomal')(() => (
    <BasicButton onClick={action('click')} disabled={boolean('disabled', false)}>
      {text('text', 'ボタン')}
    </BasicButton>
  )),
);

stories.add(
  'red',
  withInfo('BasicButton/red')(() => (
    <BasicButton red onClick={action('click')} disabled={boolean('disabled', false)}>
      {text('text', 'ボタン')}
    </BasicButton>
  )),
);

stories.add(
  'small',
  withInfo('BasicButton/small')(() => (
    <BasicButton small onClick={action('click')} disabled={boolean('disabled', false)}>
      {text('text', 'ボタン')}
    </BasicButton>
  )),
);

stories.add(
  'xsmall',
  withInfo('BasicButton/xsmall')(() => (
    <BasicButton xsmall onClick={action('click')} disabled={boolean('disabled', false)}>
      {text('text', 'ボタン')}
    </BasicButton>
  )),
);
