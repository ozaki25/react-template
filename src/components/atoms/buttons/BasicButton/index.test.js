import React from 'react';
import { snapshot } from 'test/helpers';
import BasicButton from '.';

const props = {
  onClick: jest.fn(),
};

snapshot('BasicButton/nomal', <BasicButton {...props}>ボタン</BasicButton>);
snapshot(
  'BasicButton/disabled',
  <BasicButton {...props} disabled>
    ボタン
  </BasicButton>,
);
snapshot(
  'BasicButton/red',
  <BasicButton {...props} red>
    ボタン
  </BasicButton>,
);
snapshot(
  'BasicButton/small',
  <BasicButton {...props} small>
    ボタン
  </BasicButton>,
);
snapshot(
  'BasicButton/xsmall',
  <BasicButton {...props} xsmall>
    ボタン
  </BasicButton>,
);
