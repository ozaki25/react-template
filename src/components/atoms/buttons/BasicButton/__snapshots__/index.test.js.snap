// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: BasicButton/disabled 1`] = `
.c0 {
  background-color: #333;
  border-radius: 4px;
  color: #fff;
  font-size: 16px;
  font-weight: bold;
  height: 44px;
  width: 300px;
}

<BasicButton
  disabled={true}
  onClick={[Function]}
  red={false}
  small={false}
  xsmall={false}
>
  <styled.button
    disabled={true}
    onClick={[Function]}
    red={false}
    small={false}
    xsmall={false}
  >
    <button
      className="c0"
      disabled={true}
      onClick={[Function]}
    >
      ボタン
    </button>
  </styled.button>
</BasicButton>
`;

exports[`should match snapshot: BasicButton/nomal 1`] = `
.c0 {
  background-color: #333;
  border-radius: 4px;
  color: #fff;
  font-size: 16px;
  font-weight: bold;
  height: 44px;
  width: 300px;
}

<BasicButton
  disabled={false}
  onClick={[Function]}
  red={false}
  small={false}
  xsmall={false}
>
  <styled.button
    disabled={false}
    onClick={[Function]}
    red={false}
    small={false}
    xsmall={false}
  >
    <button
      className="c0"
      disabled={false}
      onClick={[Function]}
    >
      ボタン
    </button>
  </styled.button>
</BasicButton>
`;

exports[`should match snapshot: BasicButton/red 1`] = `
.c0 {
  background-color: #e60000;
  border-radius: 4px;
  color: #fff;
  font-size: 16px;
  font-weight: bold;
  height: 44px;
  width: 300px;
}

<BasicButton
  disabled={false}
  onClick={[Function]}
  red={true}
  small={false}
  xsmall={false}
>
  <styled.button
    disabled={false}
    onClick={[Function]}
    red={true}
    small={false}
    xsmall={false}
  >
    <button
      className="c0"
      disabled={false}
      onClick={[Function]}
    >
      ボタン
    </button>
  </styled.button>
</BasicButton>
`;

exports[`should match snapshot: BasicButton/small 1`] = `
.c0 {
  background-color: #333;
  border-radius: 4px;
  color: #fff;
  font-size: 16px;
  font-weight: bold;
  height: 44px;
  width: 250px;
}

<BasicButton
  disabled={false}
  onClick={[Function]}
  red={false}
  small={true}
  xsmall={false}
>
  <styled.button
    disabled={false}
    onClick={[Function]}
    red={false}
    small={true}
    xsmall={false}
  >
    <button
      className="c0"
      disabled={false}
      onClick={[Function]}
    >
      ボタン
    </button>
  </styled.button>
</BasicButton>
`;

exports[`should match snapshot: BasicButton/xsmall 1`] = `
.c0 {
  background-color: #333;
  border-radius: 4px;
  color: #fff;
  font-size: 16px;
  font-weight: bold;
  height: 44px;
  width: 150px;
}

<BasicButton
  disabled={false}
  onClick={[Function]}
  red={false}
  small={false}
  xsmall={true}
>
  <styled.button
    disabled={false}
    onClick={[Function]}
    red={false}
    small={false}
    xsmall={true}
  >
    <button
      className="c0"
      disabled={false}
      onClick={[Function]}
    >
      ボタン
    </button>
  </styled.button>
</BasicButton>
`;
